### DISCUSSION TITLE
1亿图片的相似搜索实现

### DISCUSSION THREAD

**Discussion by jilieryuyi** [milvus-io/milvus#4903] [Q&A and General discussion]
本人电脑 windows 10 专业版移动工作站！
内存32G
英伟达gpu+8G显存

安装milvus v1.0 cpu docker运行！

对pic_search升级改造，去掉了 cache模块，所有的图片都是网络url！改造为自动网络获取训练！
查询得到id后去数据库匹配对应的url显示结果！

图片数据库待训练数据1000万！
已训练14万，电脑内存使用率已到达79%！

1亿图片该怎么实现，或者该需要什么样的硬件配置了？ 还是哪里还有什么可以优化的点？

求指点！！

**Comment by yhmo** [comment]
图片提取出来的向量是多少维度的？
milvus使用的内存量主要跟向量的维度和数量以及索引类型有关。训练用的程序也会消耗内存。
假设你提取出的向量是512维，那么1亿张图片就有1亿条向量，每条向量512*4=2048 bytes，1亿条就是200GB，如果使用IVFSQ8索引，大约就是60GB。如果要把数据全部放在内存里计算以获取最好性能的话，假设你使用IVFSQ8索引，那你就需要一台内存在64GB以上的机器，最好内存还得有些冗余，以防其他程序占用内存或者计算时峰值撑爆内存，那么最好有96GB内存。
另外CPU的性能也会影响查询速度，最好能有核数较多的CPU，好歹10核以上吧。。
